Modules/DataCollection/classes/Fields/Base/class.ilDclBaseFieldModel.php

```diff
@@ -570,7 +566,7 @@ public function checkValidity($value, ?int $record_id = null): bool
     /**
      * @param mixed $value
      */
-    protected function normalizeValue($value): string
+    protected function normalizeValue($value): ?string
```

Comment:
I would prefer to drop the return typing all together. Looking at that release7 code,  the intention was any value that is not a string gets passed through this method without being altered. In release8 anything that is not an string does not get checked but gets returned as a string. That also means that if it gets called with anything like an array an exception will be thrown